src/socket.ts

```diff
@@ -123,7 +125,8 @@ export class ConstellationSocket extends EventEmitter {
         this.on('open', () => this.schedulePing());
 
         this.on('event:hello', () => {
-            if (this.state !== State.Connecting) { // may have been closed just now
+            // may have been closed just now
+            if (this.state !== State.Connecting && this.state !== State.Reconnecting) {
                 return;
             }
```

Comment:
In retrospect this isn't necessary; websocket messages will not be sent after `socket.close()` is called. You can 🔥  this block :)